ggml-cuda.cu

```diff
@@ -6256,7 +6308,7 @@ static to_fp16_cuda_t ggml_get_to_fp16_cuda(ggml_type type) {
         case GGML_TYPE_Q5_1:
             return dequantize_block_cuda<QK5_1, QR5_1, dequantize_q5_1>;
         case GGML_TYPE_Q8_0:
-            return dequantize_block_cuda<QK8_0, QR8_0, dequantize_q8_0>;
+            return dequantize_block_q8_0_f16_cuda;
```

Comment:
Could we fallback to the previous implementation for pascal? Otherwise it is just going to crash.

Reply:
Firstly, for Pascal the new implementation works (but with terrible performance for all Pascal GPUs except for the P100). Only for Maxwell and older `bad_arch` is called.

Secondly, the `__CUDA_ARCH__` macro is unfortunately only defined for device code. So this would require a runtime check of compute capability somewhere in the CPU code. And such a check is already done in `ggml_cuda_mul_mat`. Maxwell does not have FP16 compute at all so there really aren't any circumstances where you would want to call either version of the code.

Reply:
Can't we check the cc of the current GPU? It seems reasonable to require that the current CPU is set when calling `ggml_get_to_fp16_cuda`.

Reply:
While I know that this won't happen in llama.cpp, I just want to avoid issues in the future if for example we use this function to implement dequantize with `ggml_cpy`.

Reply:
Good point.